executor/join_test.go

```diff
@@ -787,6 +787,22 @@ func (s *testSuite2) TestSubquery(c *C) {
 	tk.MustExec("insert into t2 values(1)")
 	tk.MustQuery("select * from t1 where a in (select a from t2)").Check(testkit.Rows("1"))
 
+	tk.MustExec("insert into t2 value(null)")
+	tk.MustQuery("select * from t1 where 1 in (select b from t2)").Check(testkit.Rows("1"))
+	tk.MustQuery("select * from t1 where 1 not in (select b from t2)").Check(testkit.Rows())
+	tk.MustQuery("select 1 in (select b from t2) from t1").Check(testkit.Rows("1"))
+	tk.MustQuery("select 1 in (select 1 from t2) from t1").Check(testkit.Rows("1"))
+	tk.MustQuery("select 1 not in (select b from t2) from t1").Check(testkit.Rows("0"))
+	tk.MustQuery("select 1 not in (select 1 from t2) from t1").Check(testkit.Rows("0"))
+
+	tk.MustExec("delete from t2 where b=1")
+	tk.MustQuery("select 1 in (select b from t2) from t1").Check(testkit.Rows("<nil>"))
+	tk.MustQuery("select 1 not in (select b from t2) from t1").Check(testkit.Rows("<nil>"))
+	tk.MustQuery("select 1 not in (select 1 from t2) from t1").Check(testkit.Rows("0"))
+	tk.MustQuery("select 1 in (select 1 from t2) from t1").Check(testkit.Rows("1"))
+	// TODO: this query will cause an index out of range panic
+	// tk.MustQuery("select 1 not in (select null from t1) from t2").Check(testkit.Rows())
```

Comment:
Have we checked `null in (select ...)` and `null not in (select ...)` ?

Reply:
No,because this will casue a panic when execute join(hash join,SMJ,NLJ. all).I will try to fix it later.

Reply:
I test the query before this PR:
```
mysql> select 1 in (select null from t1) from t;
+----------------------------+
| 1 in (select null from t1) |
+----------------------------+
|                          0 |
+----------------------------+
1 row in set (0.00 sec)

mysql> select 1 not in (select null from t1) from t;
+--------------------------------+
| 1 not in (select null from t1) |
+--------------------------------+
|                              1 |
+--------------------------------+
1 row in set (0.01 sec)
```
Is the panic related to this PR?

Reply:
Try this sql `select a in (select 0)xx from (select null as a)x`. This will cause panic in master because join can't hadler 'other cond' correctly. 
After this pr,this sql will generate a 'other cond'.